lib/matplotlib/axes/_axes.py

```diff
@@ -4047,8 +4060,6 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
                                        labels=tick_labels, autorange=autorange)
         if notch is None:
             notch = mpl.rcParams['boxplot.notch']
-        if vert is None:
-            vert = mpl.rcParams['boxplot.vertical']
```

Comment:
We need a deprecation on `mpl.rcParams['boxplot.vertical']` as well. This is a bit tricky.

1) We don't have a direct mechanism for deprecating `rcParams`. Likely a custom validator similar to https://github.com/matplotlib/matplotlib/blob/8fb842c78f1b57b13a3d0fe63e83f3d31694f887/lib/matplotlib/rcsetup.py#L441 is the way to go.
2) Determining priority: Not completely thought through, but I believe, if `mpl.rcParams['boxplot.vertical'] = False` that should take precedence over `orientation`. `mpl.rcParams['boxplot.vertical'] = True` should likely be ignored. That should give expected results for reasonable use cases. - We cannot easily detect whether the rcParam and `orientation` were explicitly set or use their defaults. That detection would require extra machinery (i.e. an extra None=not set state), which we probably can get away without folling the above rule.

Reply:
We do have some rcParams deprecation mechanism here:
https://github.com/matplotlib/matplotlib/blob/ec0653d9f54ca5c2ae6d5e8f84daaed8fe178886/lib/matplotlib/__init__.py#L637-L646